git-worktree/src/lib.rs

```diff
@@ -1 +1,95 @@
 #![forbid(unsafe_code, rust_2018_idioms)]
+//! Git Worktree
+
+use anyhow::Result;
+use git_hash::oid;
+use git_index::{
+    entry::{Flags, Mode},
+    State,
+};
+use git_object::bstr::ByteSlice;
+use git_object::Data;
+use std::convert::TryFrom;
+use std::fs;
+use std::fs::create_dir_all;
+use std::path::Path;
+use std::time::SystemTime;
+
+#[cfg(unix)]
+use std::os::unix::fs::PermissionsExt;
+
+/// Copy index to `path`
+pub fn copy_index<P, Find>(state: &mut State, path: P, mut find: Find, opts: Options) -> Result<()>
+where
+    P: AsRef<Path>,
+    Find: for<'a> FnMut(&oid, &'a mut Vec<u8>) -> Option<Data<'a>>,
+{
+    let path = path.as_ref();
+    let paths = state
+        .entries()
+        .iter()
+        .map(|e| e.path(state).to_path().map(|p| p.to_path_buf()))
+        .collect::<Result<Vec<_>, _>>()?;
+    let mut buf = Vec::new();
+    for (i, entry) in state.entries_mut().iter_mut().enumerate() {
+        if entry.flags.contains(Flags::SKIP_WORKTREE) {
+            continue;
+        }
+        let time = SystemTime::now().duration_since(SystemTime::UNIX_EPOCH)?;
```

Comment:
This must be the time returned by a `metadata()` call, as it must match what's actually on disk. Otherwise git will think the file changed.